prometheus/src/banks_with_commitments.rs

```diff
@@ -59,11 +59,14 @@ impl BanksWithCommitments {
     }
 
     /// Call function callback for each commitment level, and returns a vector of metrics.
-    pub fn for_each_commitment<F: Fn(&Bank) -> Metric>(&self, get: F) -> Vec<Metric> {
-        vec![
-            get(&self.finalized_bank).with_label("commitment_level", "finalized".to_owned()),
-            get(&self.confirmed_bank).with_label("commitment_level", "confirmed".to_owned()),
-            get(&self.processed_bank).with_label("commitment_level", "processed".to_owned()),
-        ]
+    pub fn for_each_commitment<F: Fn(&Bank) -> Option<Metric>>(
+        &self,
+        get: F,
+    ) -> Option<Vec<Metric>> {
+        Some(vec![
+            get(&self.finalized_bank)?.with_label("commitment_level", "finalized".to_owned()),
+            get(&self.confirmed_bank)?.with_label("commitment_level", "confirmed".to_owned()),
+            get(&self.processed_bank)?.with_label("commitment_level", "processed".to_owned()),
+        ])
```

Comment:
We can do it like this:

```suggestion
    ) -> Vec<Metric> {
        let mut result = Vec::with_capacity(3);
        result.extend(get(&self.finalized_bank).map(|m| m.with_label("commitment_level", "finalized".to_owned()));
        result.extend(get(&self.confirmed_bank).map(|m| m.with_label("commitment_level", "confirmed".to_owned()));
        result.extend(get(&self.processed_bank).map(|m| m.with_label("commitment_level", "processed".to_owned()));
        result
```

It doesn’t have to be all-or-nothing for the metrics.

Reply:
But thinking about it some more ... instead of returning an `Option<Metric>`, shouldn’t we maybe return a `Vec<Metric>` from the closure? If we are going to monitor multiple vote accounts, it will be the same metric, but with different labels. So inside the closure we would loop over the vote account addresses, and append to the vec.

Hmm, and then the closure has to allocate the vec and put things in there, and then we deconstruct the vec again and concatenate it here ... we could do better by defining a wrapper type:

```rust
struct MetricsVec {
    inner: Vec<Metric>,
    commitment_level: String,
}

impl MetricVec {
    pub fn push(&mut self, metric: Metric) {
        self.inner.push(metric.with_label("commitment_level", self.commitment_level.clone()));
    }
}
```
and then the closure would change to `F: Fn(&Bank, &mut MetricsVec)`, and it could put its metrics directly into the right vec. If there are multiple, it can add multiple. If there are none, it just doesn’t push anything.